rfcs/0039-minaform-tool-design.md

```diff
@@ -0,0 +1,143 @@
+## Summary
+[summary]: #summary
+
+This rfc proposes a new tool dubbed "minaform" for reliably creating and managing QA nets and production deployments. Our existing setup is very hacky and requires knowledge of many tools to use effectively. The intention with minaform is to build a flexible enough tool that resolves major pain points with the existing setup, and standardizes a single tool for folks to learn/understand.
+
+## Motivation
+[motivation]: #motivation
+
+The goal is to standardize community-oriented and internal qa networks with a simple tool to ensure that the process for creating and deploying a new testnet is code-reviewed, documented, and standard across all deployment scenarios. As of this RFC, additional requirements for community members themselves or external partners to make use of this tooling is out-of-scope but the goal is that this implementation will be easier to maintain and extend as needs arise.
+
+The following issues with the existing system were reported by at least one person as "pain points" that consume time/debugging effort when deploying networks:
```

Comment:
Have to agree w/ Nathan on the idea that all aspects of the deployment process should now be expressible in Terraform with `terraform apply/destroy` providing a simple interface for triggering proper standup and destruction of testnet environments with a well-developed dependency management system in place and proper deployment ordering handled. While I like the idea of creating a Docker image wrapper around deployment tooling (to manage versions, consistency etc), honestly solutions to each of the other pain points listed may be better suited as terraform resources. Whether that comes in the form of `local-execs` to execute scripts for key and ledger generation tasks or, better yet, kubernetes jobs which to Nathan's point could help offload the burden on developer's machines for handling the processing necessary to generate these artifacts I'm not sure I can get onboard with looking outside of terraform for this and introducing yet another bash script into the process. Especially considering the potential breadth of usage of this tool throughout engineering if not all of O(1) and the major flaws bash has when it comes to user-interfaces.